### DISCUSSION TITLE
Unable to use brew to install other software

### DISCUSSION THREAD

**Discussion by saikongara** [Homebrew/discussions#151] [Everyday usage] [answered]
Unable to use brew to install other software like azure CLI.

The following directories are not writable by your user:
/usr/local/share/man/man5

Tried to re-install the brew, couldn't have sudo access, which I believe brew doesn't require.

Need suggestions to get this working.

**Comment by gromgit** [comment]
Who installed Homebrew on your machine? If someone else did the install, all the relevant directories under `/usr/local` would be owned by them, which would explain why you can't write to them.

If that were the case, then the simplest solution is for them to run:
```
sudo chown -R <your_username>:admin /usr/local/*
```
And yes, sudo access isn't required for normal Homebrew operations, but the Homebrew install script _does_ require it to set up the `/usr/local` hierarchy to be writable by you.

**Reply by saikongara** [reply]
Is there any other way instead of using sudo access to fix this issue. Even I tried to re-install Brew, but it is asking for **sudo** access which in normal cases is not required !

**Reply by gromgit** [reply]
> Is there any other way instead of using sudo access to fix this issue

If ordinary users like you can change ownerships and permissions of stuff not owned by you, and without yourself requiring special permissions, you'll have a **security nightmare** on your hands. Not gonna happen.

That said, if you can't convince your admin to run that _one single command_ I mentioned above, you _could_ [reinstall Homebrew under some other directory that you own](https://docs.brew.sh/Installation#untar-anywhere). However, note that none of the Homebrew formulae are tested for installation under arbitrary directories like `/Users/saikongara/your/special/dir`, so you may experience random breakage that you'll have to solve on your own.

The choice is yours.

**Comment by SMillerDev** [comment]
> Even I tried to re-install Brew, but it is asking for sudo access which in normal cases is not required !

The brew install always asked/asks for sudo access exactly so it doesn't have to do that in normal operations

**Comment by NOTiFYcouk** [comment]
Fixed @ https://apple.stackexchange.com/questions/401899/homebrew-your-clt-does-not-support-macos-11-0

Simplest solution that worked when upgrading to Big Sur. A clean install may not require the solution below. I believe there is some inconsistency with the way CLT is upgraded.

sudo rm -rf /Library/Developer/CommandLineTools
sudo xcode-select --install

**Reply by SMillerDev** [reply]
That can't possibly fix this issue, please only post thingst that actually fix the issue presented

**Comment by saikongara** [comment, marked as answer]
got the Admin rights and ran `sudo chown -R <your_username>:admin /usr/local/*` and the issue resolved

**Comment by tresf** [comment]
Are [user-space installs](https://superuser.com/a/1025497/443147) still discouraged?  This seems like a reasonable answer to anyone without `sudo`, assuming the formula don't falter as a result.

**Reply by carlocab** [reply]
From https://docs.brew.sh/Installation#untar-anywhere

>However do yourself a favour and install to /usr/local on macOS Intel, /opt/homebrew on macOS ARM, and /home/linuxbrew/.linuxbrew on Linux. Some things may not build when installed elsewhere. One of the reasons Homebrew just works relative to the competition is because we recommend installing here. Pick another prefix at your peril!
